### ISSUE TITLE
Patch noarch python packages to require python 3

### ISSUE THREAD

**Issue by mariusvniekerk** [conda-forge/conda-forge-repodata-patches-feedstock#93]
Should we patch all python noarch package built after some date (say 2020-10-01) to require python 3?

This should avoid issues with having modern noarch packages being installable in py2 whilst not working.

**Comment by isuruf** [comment]
I think we can do that if there's no python constraint there. We should first add a lint to add a lower bound on python and maybe also a mini migrator to add `python >=3.6`

**Comment by mariusvniekerk** [comment]
In favor of all of that

**Comment by betteridiot** [comment]
What happens here if a package is explicitly written to work on 2.7 and greater? My code was just linted https://github.com/conda-forge/bamnostic-feedstock/pull/36 and the co-maintainer made the `python >= 3.6` update. I was wondering if this update now prevents the package from installing on `conda` environments using 2.7?

**Comment by isuruf** [comment]
Yes, the new packages cannot be installed in 2.7, but the old packages can be. If you want to avoid the lint, you can add `python >=2.7`.
